File: angular-app/src/app/core/application/decorator/helpers/strategies/help-strategy.ts
import { UserAction } from '../../../enums/user-action';
import { DecoratorTextHelper } from '../decorator-text-helper';
import { BaseHelperStrategy } from './abstracts/base-helper-strategy';

const phrases = ['help', 'ajuda', 'comandos'];

export class HelpStrategy extends BaseHelperStrategy {
  override priority = 5;

  override canHandle(normalized: string, tokens: string[]): boolean {
    return DecoratorTextHelper.bestMatch(normalized, phrases, 0.75) !== null;
  }

  override handle(normalized: string, tokens: string[]): [UserAction, string] {
    return [UserAction.MESSAGE, 'help'];
  }
}

File: python-app/application/decorator/strategies/help_strategy.py
from collections.abc import Sequence
from application.enums.user_action import UserAction
from application.decorator.helpers.decorator_text_helper import DecoratorTextHelper
from application.decorator.strategies.abstracts.base_helper_strategy import BaseHelperStrategy
from application.decorator.strategies.decorator.decorator_helper_strategy import strategy

@strategy(priority=5)
class HelpStrategy(BaseHelperStrategy):
    _PHRASES = ["help", "ajuda", "comandos"]

    def can_handle(self, normalized: str, tokens: Sequence[str]) -> bool:
        return DecoratorTextHelper.best_match(normalized, self._PHRASES, 0.75) is not None

    def handle(self, normalized: str, tokens: Sequence[str]):
        return UserAction.MESSAGE, "help"
